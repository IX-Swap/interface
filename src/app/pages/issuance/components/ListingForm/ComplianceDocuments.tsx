import React from 'react'
import { Grid, Typography, Link } from '@mui/material'
import { API_URL } from 'config'
import { FormSectionHeader } from 'ui/FormSectionHeader/FormSectionHeader'
import { UploadDocumentField } from 'app/pages/identity/components/UploadDocumentsForm/UploadDocumentField/UploadDocumentField'

export const ComplianceDocuments = () => {
  return (
    <Grid item container direction='column' spacing={5}>
      <Grid item>
        <FormSectionHeader title={'Upload Documents'} />
      </Grid>

      <Grid item container direction={'column'} spacing={1.5}>
        <Grid item>
          <Typography sx={{ color: '#778194', marginTop: '-20px' }}>
            Please upload your compliance documents based on the{' '}
            <Link
              href={`${API_URL}/IXExchangeTemplateListingProfile.pdf`}
              style={{ cursor: 'pointer' }}
              sx={{
                whiteSpace: 'nowrap'
              }}
              target='_blank'
            >
              IX Exchange Listing Profile Template
            </Link>
            .
          </Typography>
        </Grid>

        <Grid item mt={3}>
          <UploadDocumentField
            name='documents'
            label='Compliance Documents'
            helperElement={
              <Typography variant='body1' sx={{ color: '#778194' }}>
                Key Information Summary, Letter from the Chairperson, Issuer and
                What It Does, Key Features of the Issuer&apos;s Digital
                Securities, Issuer&apos;s Financial Information, Risks to The
                Issuer&apos;s Business and Plans, Tax, Where You Can Find More
                Information, Contact Information, Timetable of Relevant Dates
              </Typography>
            }
          />
        </Grid>
      </Grid>
    </Grid>
  )
}
