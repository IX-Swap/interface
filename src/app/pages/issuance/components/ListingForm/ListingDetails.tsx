import {
  Button,
  FormControl,
  FormControlLabel,
  Grid,
  RadioGroup,
  Typography
} from '@mui/material'
import { useStyles } from 'app/pages/accounts/components/CurrencySelect/CurrencySelect.styles'
import { FormSectionHeader } from 'app/pages/identity/components/FormSectionHeader'
import { DSOSelect } from 'app/pages/issuance/components/IssuanceLanding/DSOSelect'
import { useDSOsByUserId } from 'app/pages/issuance/hooks/useDSOsByUserId'
import classnames from 'classnames'
import { UIRadio } from 'components/UIRadio/UIRadio'
import React, { useState } from 'react'

export type ListingType = 'Secondary' | 'Otc' | 'Both'

export interface ListingDetailsProps {
  listingType: ListingType
  setListingType: (value: ListingType) => void
  onImportClick: (value: string) => void
}

export const radioButtonsList = [
  {
    label: 'Exchange',
    value: 'Secondary'
  },
  {
    label: 'OTC',
    value: 'Otc'
  },
  {
    label: 'Both',
    value: 'Both'
  }
]

export const ListingDetails = ({
  listingType,
  setListingType,
  onImportClick
}: ListingDetailsProps) => {
  const classes = useStyles()

  const { data, isLoading } = useDSOsByUserId('Approved')
  const [selectedDSOValue, setSelectedDSOValue] = useState<string>('')

  return (
    <Grid item container direction={'column'} spacing={{ xs: 4, md: 5 }}>
      <Grid item>
        <FormSectionHeader title={'Listing details'} />
      </Grid>

      <Grid item container direction={'column'} spacing={1.5}>
        <Grid item>
          <Typography>Where do you want to list this offering?</Typography>
        </Grid>

        <Grid item>
          <RadioGroup name={'listPlace'} value={listingType}>
            <Grid
              container
              display={'grid'}
              gridTemplateColumns={{ xs: '1fr', md: '1fr 1fr 1fr' }}
              gap={1.5}
            >
              {radioButtonsList.map(({ label, value }) => {
                return (
                  <Grid
                    item
                    data-testid={'buttonWrapper'}
                    className={classnames(classes.button, {
                      [classes.active]: listingType === value
                    })}
                    onClick={() => {
                      setListingType(value as ListingType)
                    }}
                  >
                    <FormControlLabel
                      label={label}
                      value={value}
                      control={<UIRadio />}
                    />
                  </Grid>
                )
              })}
            </Grid>
          </RadioGroup>
        </Grid>
      </Grid>

      <Grid item container direction={'column'} spacing={1.5}>
        <Grid item>
          <Typography>Import data from issuance</Typography>
        </Grid>

        <Grid
          item
          container
          alignItems={{ xs: 'normal', md: 'center' }}
          direction={{ xs: 'column', md: 'row' }}
        >
          <Grid
            item
            flexGrow={1}
            marginRight={{ xs: 0, md: 1.5 }}
            marginBottom={{ xs: 4, md: 0 }}
          >
            <FormControl fullWidth variant='outlined'>
              <DSOSelect
                style={{ height: 50 }}
                fullWidth
                label='My DSO'
                value={selectedDSOValue}
                disabled={isLoading}
                options={data.list}
                onChange={value =>
                  setSelectedDSOValue(value.target.value as string)
                }
              />
            </FormControl>
          </Grid>
          <Grid item>
            <Button
              sx={{
                height: 50,
                width: {
                  xs: '100%',
                  md: 123
                }
              }}
              color={'primary'}
              variant={'contained'}
              onClick={() => {
                if (selectedDSOValue.length > 0) {
                  onImportClick(selectedDSOValue)
                }
              }}
            >
              Import
            </Button>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  )
}
