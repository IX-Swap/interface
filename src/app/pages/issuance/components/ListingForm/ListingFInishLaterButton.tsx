import { Button } from '@mui/material'
import { getIdFromObj } from 'helpers/strings'
import React from 'react'
import { useFormContext } from 'react-hook-form'
import { DigitalSecurityOffering } from 'types/dso'
import { Listing, ListingFormValues } from 'app/pages/issuance/types/listings'
import { getUpdateListingPayload } from 'app/pages/issuance/utils/listing'
import { useCreateListing } from 'app/pages/issuance/hooks/useCreateListing'
import { useUpdateListing } from 'app/pages/issuance/hooks/useUpdateListing'

export interface ListingFinishLaterButtonProps {
  listing: DigitalSecurityOffering | Listing | undefined
  isDataFromDSO: boolean
}

export const ListingFinishLaterButton = (
  props: ListingFinishLaterButtonProps
) => {
  const { listing, isDataFromDSO } = props
  const listingId = getIdFromObj(listing)

  const { watch } = useFormContext<ListingFormValues>()
  const [createListing, { isLoading: isCreating }] = useCreateListing()
  const [updateListing, { isLoading: isUpdating }] = useUpdateListing(
    listingId,
    typeof listing?.user === 'string'
      ? listing?.user
      : getIdFromObj(listing?.user) ?? listing?.createdBy ?? ''
  )
  const formValues = getUpdateListingPayload({
    ...watch(),
    status: 'Draft'
  } as any)

  const handleClick =
    listing === undefined || isDataFromDSO
      ? async () => await createListing(formValues)
      : async () => await updateListing(formValues)

  return (
    <Button
      variant='outlined'
      color='primary'
      onClick={handleClick}
      disabled={isCreating || isUpdating}
    >
      {listing === undefined || isDataFromDSO ? 'Save Draft' : 'Save'}
    </Button>
  )
}
