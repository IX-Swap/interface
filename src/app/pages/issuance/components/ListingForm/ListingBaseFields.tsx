import { Box, Grid } from '@mui/material'
import { FormSectionHeader } from 'app/components/DSO/components/FormSectionHeader'
import { documentValueExtractor } from 'app/components/DSO/utils'
import { ListingHiddenFields } from 'app/pages/issuance/components/ListingForm/ListingHiddenFields'
import { initialListingFormValues } from 'app/pages/issuance/consts/listing'
import { Dropzone } from 'components/dataroom/Dropzone'
import { CapitalStructureSelect } from 'components/form/CapitalStructureSelect'
import { CorporateSelect } from 'components/form/CorporateSelect'
import { NetworkSelect } from 'components/form/NetworkSelect'
import { NumericInput } from 'components/form/NumericInput'
import { TypedField } from 'components/form/TypedField'
import { DateTimePicker } from 'components/form/_DateTimePicker'
import { DataroomFileType } from 'config/dataroom'
import { positiveNumberFormat } from 'config/numberFormat'
import { dateTimeValueExtractor, numericValueExtractor } from 'helpers/forms'
import React from 'react'
import { useFormContext } from 'react-hook-form'
import { TextInput } from 'ui/TextInput/TextInput'
export interface ListingBaseFieldsProps {
  isNew: boolean
  isLive: boolean
  isDataFromDSO: boolean
}

export const ListingBaseFields = (props: ListingBaseFieldsProps) => {
  const { isNew, isLive, isDataFromDSO } = props
  const { control, watch } = useFormContext()

  return (
    <Grid item>
      <Grid container direction='column' spacing={2}>
        <Grid item>
          <FormSectionHeader title='General Information' />
        </Grid>
        <Grid item>
          <TypedField
            customRenderer
            component={Dropzone}
            name='logo'
            label='Upload Logo'
            control={control}
            valueExtractor={documentValueExtractor}
            accept={DataroomFileType.image}
            documentInfo={{
              type: 'DSO Logo'
            }}
          />
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={6}>
              <TypedField
                control={control}
                component={CapitalStructureSelect}
                label='Capital Structure'
                name='capitalStructure'
                helperText='Offering terms will be changed based on your capital structure'
                variant='outlined'
                inputProps={{ 'data-testid': 'capital-structure' }}
                disabled={
                  isDataFromDSO &&
                  watch('capitalStructure') !==
                    initialListingFormValues.capitalStructure
                }
              />
            </Grid>
            <Grid item xs={12} md={6}>
              <TypedField
                component={NetworkSelect}
                label='Network'
                name='network'
                disabled={
                  !isNew ||
                  (isDataFromDSO &&
                    watch('network') !== initialListingFormValues.network)
                }
                control={control}
                helperText='Select your blockchain network from the list'
                variant='outlined'
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={4}>
              <TypedField
                component={TextInput}
                label='Token Name'
                name='tokenName'
                disabled={
                  isLive ||
                  (isDataFromDSO &&
                    watch('tokenName') !== initialListingFormValues.tokenName)
                }
                control={control}
                helperText='Name of the token that describes your offering the best'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={4}>
              <TypedField
                component={TextInput}
                label='Symbol'
                name='tokenSymbol'
                disabled={
                  isLive ||
                  (isDataFromDSO &&
                    watch('tokenSymbol') !==
                      initialListingFormValues.tokenSymbol)
                }
                control={control}
                helperText='Token symbol'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={4}>
              <TypedField
                component={NumericInput}
                customRenderer
                numberFormat={positiveNumberFormat}
                label='Decimal Places'
                name='decimals'
                control={control}
                helperText='Decimal Places'
                variant='outlined'
                valueExtractor={numericValueExtractor}
                fullWidth
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12}>
              <TypedField
                component={CorporateSelect}
                label='Corporate Name'
                name='corporate'
                control={control}
                helperText='Select your corporate from the list'
                variant='outlined'
                disabled={
                  isDataFromDSO &&
                  watch('corporate') !== initialListingFormValues.corporate
                }
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={6}>
              <TypedField
                component={DateTimePicker}
                customRenderer
                label='Launch Date'
                name='launchDate'
                control={control}
                disabled={
                  isLive || (isDataFromDSO && watch('launchDate') !== null)
                }
                valueExtractor={dateTimeValueExtractor}
                defaultValue={null}
                helperText='Offering launch date'
              />
            </Grid>
            <Grid item xs={12} md={6}>
              <TypedField
                component={DateTimePicker}
                customRenderer
                label='Completion Date'
                name='completionDate'
                control={control}
                valueExtractor={dateTimeValueExtractor}
                defaultValue={null}
                helperText='Offering completion date'
                disabled={isDataFromDSO && watch('completionDate') !== null}
              />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
      <Box display='none'>
        <ListingHiddenFields />
      </Box>
    </Grid>
  )
}
