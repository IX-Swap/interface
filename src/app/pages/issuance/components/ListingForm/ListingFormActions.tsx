import React from 'react'
import { Box, Button } from '@mui/material'
import { DigitalSecurityOffering } from 'types/dso'
import { useHistory, generatePath } from 'react-router-dom'
import { IssuanceRoute } from 'app/pages/issuance/router/config'
import { ListingFinishLaterButton } from 'app/pages/issuance/components/ListingForm/ListinfFInishLaterButton'
import { getIdFromObj } from 'helpers/strings'
import { Listing } from 'app/pages/issuance/types/listings'

export interface ListingFormActionsProps {
  listing: DigitalSecurityOffering | Listing | undefined
  isDataFromDSO: boolean
}

export const ListingFormActions = (props: ListingFormActionsProps) => {
  const { listing, isDataFromDSO } = props
  const { push } = useHistory()

  return (
    <>
      {!(listing === undefined || isDataFromDSO) && (
        <Button
          variant={'contained'}
          color='primary'
          disableElevation
          onClick={() =>
            push(
              generatePath(IssuanceRoute.previewListing, {
                listingId: listing?._id,
                issuerId:
                  typeof listing?.user === 'string'
                    ? listing?.user
                    : getIdFromObj(listing?.user)
              })
            )
          }
        >
          Preview
        </Button>
      )}
      <Box mx={1} component='span' />
      <ListingFinishLaterButton
        isDataFromDSO={isDataFromDSO}
        listing={listing}
      />
    </>
  )
}
