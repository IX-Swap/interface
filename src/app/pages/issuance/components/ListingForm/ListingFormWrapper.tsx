import React, { useState } from 'react'
import { Grid } from '@mui/material'
import { useDSOById } from 'app/pages/invest/hooks/useDSOById'
import {
  ListingDetails
  // ListingType
} from 'app/pages/issuance/components/ListingForm/ListingDetails'
import { LoadingIndicator } from 'app/components/LoadingIndicator/LoadingIndicator'
import { ListingForm } from 'app/pages/issuance/components/ListingForm/ListingForm'
import {
  getIdFromDSOSelectValue,
  getIssuerIdFromDSOSelectValue
} from 'app/pages/issuance/utils/utils'
import { Listing } from 'app/pages/issuance/types/listings'
import { FieldContainer } from 'app/pages/identity/components/FieldContainer/FieldContainer'

export interface ListingFormWrapperProps {
  data?: Listing
  isNew?: boolean
}

export const ListingFormWrapper = (props: ListingFormWrapperProps) => {
  const { data: initialListingData, isNew = false } = props
  // const [listingType, setListingType] = useState<ListingType>('Secondary')
  const [listingType, setListingType] = useState<any>('')
  const [dsoId, setDsoId] = useState<string | undefined>(undefined)
  const [issuerId, setIssuerId] = useState<string | undefined>(undefined)

  const { data: dsoData, isLoading } = useDSOById(dsoId, issuerId)
  const data = initialListingData ?? dsoData

  const renderListingForm = () => {
    if (
      dsoId === undefined &&
      issuerId === undefined &&
      initialListingData === undefined
    ) {
      return null
    }
    if (isLoading) {
      return <LoadingIndicator />
    }

    return (
      <Grid item>
        <ListingForm data={data} isNew={isNew} listingType={listingType} />
      </Grid>
    )
  }

  return (
    <Grid item container direction={'column'} spacing={2}>
      {!isNew ? null : (
        <Grid item>
          <FieldContainer>
            <ListingDetails
              listingType={listingType}
              setListingType={setListingType}
              onImportClick={value => {
                setDsoId(getIdFromDSOSelectValue(value))
                setIssuerId(getIssuerIdFromDSOSelectValue(value))
              }}
            />
          </FieldContainer>
        </Grid>
      )}

      {renderListingForm()}
    </Grid>
  )
}
