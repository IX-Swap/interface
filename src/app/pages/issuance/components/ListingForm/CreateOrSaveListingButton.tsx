import React from 'react'
import { Button } from '@mui/material'
import { useAuth } from 'hooks/auth/useAuth'
import { getIdFromObj } from 'helpers/strings'
import { useFormContext } from 'react-hook-form'
import { DigitalSecurityOffering } from 'types/dso'
import { getUpdateListingPayload } from 'app/pages/issuance/utils/listing'
import { useCreateListing } from 'app/pages/issuance/hooks/useCreateListing'
import { useUpdateListing } from 'app/pages/issuance/hooks/useUpdateListing'
import { Listing, ListingFormValues } from 'app/pages/issuance/types/listings'

import { ListingType } from 'app/pages/issuance/components/ListingForm/ListingDetails'

export interface CreateOrSaveListingButtonProps {
  listing: DigitalSecurityOffering | Listing | undefined
  listingType?: ListingType | any
  // listingType?: null | string | undefined
  isDataFromDSO: boolean
}

export const CreateOrSaveListingButton = (
  props: CreateOrSaveListingButtonProps
) => {
  const { user } = useAuth()
  const userId = getIdFromObj(user)
  const { listing, isDataFromDSO, listingType } = props
  const listingId = getIdFromObj(listing)

  const { watch } = useFormContext<ListingFormValues>()
  const [createListing, { isLoading: isCreating }] =
    useCreateListing(listingType)
  const [updateListing, { isLoading: isUpdating }] = useUpdateListing(
    listingId,
    typeof listing?.user === 'string'
      ? listing?.user
      : getIdFromObj(listing?.user) ?? listing?.createdBy ?? '',
    listingType
  )
  const { dso, ...defaultFormValues } = watch()
  const formValues = getUpdateListingPayload({
    ...defaultFormValues,
    type: listingType,
    dso: dso,
    userId: userId
  } as any)

  console.log({
    defaultFormValues,
    dso,
    formValues
  })
  const handleClick =
    listing === undefined || isDataFromDSO
      ? async () => await createListing(formValues)
      : async () => await updateListing(formValues)

  return (
    <Button
      variant='contained'
      color='primary'
      onClick={handleClick}
      disabled={isCreating || isUpdating}
    >
      {listing === undefined || isDataFromDSO ? 'Create Listing' : 'Save'}
    </Button>
  )
}
