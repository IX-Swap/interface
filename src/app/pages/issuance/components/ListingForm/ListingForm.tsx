import React, { useEffect, useState } from 'react'
import { DigitalSecurityOffering } from 'types/dso'
import { isDSOLive } from 'app/components/DSO/utils'
import { Grid } from '@mui/material'
import { Form } from 'components/form/Form'
import { ListingFormFields } from 'app/pages/issuance/components/ListingForm/ListingFormFields'
import { ListingFormActions } from 'app/pages/issuance/components/ListingForm/ListingFormActions'
import { Listing } from 'app/pages/issuance/types/listings'
import {
  transformDataFromDSOToListingFormValue,
  transformListingToListingFormValue
} from 'app/pages/issuance/utils/listing'
import { ListingType } from 'app/pages/issuance/components/ListingForm/ListingDetails'

export interface ListingFormProps {
  data?: DigitalSecurityOffering | Listing
  isNew?: boolean
  listingType: null | ListingType | any
}

export const ListingForm = (props: ListingFormProps) => {
  const { data, isNew = false, listingType } = props
  const isLive = isDSOLive(data as any)
  const isDataFromDSO = data !== undefined && !('maximumTradeUnits' in data)
  const [listingTypeUpdated, setListingTypeUpdated] = useState('')

  useEffect(() => {
    switch (data?.listingType) {
      case 'Exchange':
        setListingTypeUpdated('Secondary')
        break
      case 'OTC':
        setListingTypeUpdated('Otc')
        break
      case 'Exchange/OTC':
        setListingTypeUpdated('Both')
        break
    }
  }, [data?.listingType])

  return (
    <Form
      data-testid='listing-form'
      defaultValues={
        data !== undefined && 'maximumTradeUnits' in data
          ? transformListingToListingFormValue(data)
          : transformDataFromDSOToListingFormValue(data)
      }
    >
      <Grid container direction={'column'} spacing={2}>
        <Grid item>
          <ListingFormFields
            isNew={isNew}
            isLive={isLive}
            isDataFromDSO={isDataFromDSO}
            status={data?.status}
            data={data}
          />
        </Grid>

        <Grid item container justifyContent={'flex-end'}>
          <Grid item>
            <ListingFormActions
              isDataFromDSO={isDataFromDSO}
              listing={data}
              listingType={
                listingType === '' ? listingTypeUpdated : listingType
              }
            />
          </Grid>
        </Grid>
      </Grid>
    </Form>
  )
}
