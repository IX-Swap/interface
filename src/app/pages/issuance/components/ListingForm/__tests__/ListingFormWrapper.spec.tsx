import React from 'react'
import { render } from 'test-utils'
import { ListingFormWrapper } from 'app/pages/issuance/components/ListingForm/ListingFormWrapper'

describe('ListingFormWrapper', () => {
  it('should match snapshot', () => {
    const { container } = render(<ListingFormWrapper isNew />)
    expect(container).toMatchSnapshot()
  })
})
