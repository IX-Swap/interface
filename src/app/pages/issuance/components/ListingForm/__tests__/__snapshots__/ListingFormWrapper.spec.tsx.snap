// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingFormWrapper should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-10h8c1q-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1r7sl14-MuiPaper-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-4 MuiGrid-spacing-md-5 MuiGrid-direction-xs-column css-1b9jecq-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-1br8tjs-MuiTypography-root"
              >
                Listing details
              </h5>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1.5 MuiGrid-direction-xs-column css-13om8kr-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1b4l2pg-MuiTypography-root"
              >
                Where do you want to list this offering?
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <div
                class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
                role="radiogroup"
              >
                <div
                  class="MuiGrid-root MuiGrid-container css-h1rrzm-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item makeStyles-button-2 css-13i4rnv-MuiGrid-root"
                    data-testid="buttonWrapper"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-pvro6c-MuiButtonBase-root-MuiRadio-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          name="listPlace"
                          type="radio"
                          value="Secondary"
                        />
                        <span
                          class="css-hyxlzm"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1vkanhx-MuiSvgIcon-root"
                            data-testid="RadioButtonUncheckedIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                            />
                          </svg>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ekoonk-MuiSvgIcon-root"
                            data-testid="RadioButtonCheckedIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1b4l2pg-MuiTypography-root"
                      >
                        Exchange
                      </span>
                    </label>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item makeStyles-button-2 css-13i4rnv-MuiGrid-root"
                    data-testid="buttonWrapper"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-pvro6c-MuiButtonBase-root-MuiRadio-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          name="listPlace"
                          type="radio"
                          value="Otc"
                        />
                        <span
                          class="css-hyxlzm"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1vkanhx-MuiSvgIcon-root"
                            data-testid="RadioButtonUncheckedIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                            />
                          </svg>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ekoonk-MuiSvgIcon-root"
                            data-testid="RadioButtonCheckedIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1b4l2pg-MuiTypography-root"
                      >
                        OTC
                      </span>
                    </label>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item makeStyles-button-2 css-13i4rnv-MuiGrid-root"
                    data-testid="buttonWrapper"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiRadio-root MuiRadio-colorPrimary MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root css-pvro6c-MuiButtonBase-root-MuiRadio-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          name="listPlace"
                          type="radio"
                          value="Both"
                        />
                        <span
                          class="css-hyxlzm"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1vkanhx-MuiSvgIcon-root"
                            data-testid="RadioButtonUncheckedIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                            />
                          </svg>
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-ekoonk-MuiSvgIcon-root"
                            data-testid="RadioButtonCheckedIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1b4l2pg-MuiTypography-root"
                      >
                        Both
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1.5 MuiGrid-direction-xs-column css-13om8kr-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1b4l2pg-MuiTypography-root"
              >
                Import data from issuance
              </p>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-[object Object] css-1o6rls1-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-1mygp55-MuiGrid-root"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                >
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl makeStyles-wrapper-4 css-sr8rm-MuiInputBase-root-MuiOutlinedInput-root"
                    style="height: 50px;"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="makeStyles-select-5 MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-1ub9as-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                      role="button"
                      tabindex="0"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </div>
                    <input
                      aria-hidden="true"
                      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      tabindex="-1"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-7glybp-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-nnbavb"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-hv1m69-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Import
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
